<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Construindo uma árvore de máxima verossimilhança com o software IQ-TREE3 | </title> <meta name="author" content="Pedro P. G. Taucci"> <meta name="description" content="A step-by-step beginner's guide"> <meta name="keywords" content="herpetology, amphibians, reptiles, systematics, phylogeography"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="/assets/img/logos/favicon.svg?964134e30467bb50cb172b62f5c095b4"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://pedrotaucce.github.io/blog/2025/maximum_likelihood_tree/"> <script src="/assets/js/theme.js?9a0c749ec5240d9cda97bc72359a72c0"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>initTheme();</script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand" href="/"> <img id="logo-img" src="/assets/img/logos/logo_verde.png" alt="Logo" style="width: 100px; height: auto;"> </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">Home </a> </li> <li class="nav-item active"> <a class="nav-link" href="/blog/">Blog </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">Publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">Projects </a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">Repositories </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">CV </a> </li> <li class="nav-item "> <a class="nav-link" href="/teaching/">Teaching </a> </li> <li class="nav-item "> <a class="nav-link" href="/people/">People </a> </li> <li class="nav-item"> <button id="search-toggle" title="Search" onclick="openSearchModal()"> <span class="nav-link">ctrl k <i class="ti ti-search"></i></span> </button> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <script>document.addEventListener("DOMContentLoaded",function(){function t(){const t=document.documentElement.getAttribute("data-theme");e.src="dark"===t?"/assets/img/logos/logo_laranja.png":"/assets/img/logos/logo_verde.png"}const e=document.getElementById("logo-img");new MutationObserver(t).observe(document.documentElement,{attributes:!0,attributeFilter:["data-theme"]}),t()});</script> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">Construindo uma árvore de máxima verossimilhança com o software IQ-TREE3</h1> <p class="post-meta"> Created in May 28, 2025 </p> <p class="post-tags"> <a href="/blog/2025"> <i class="fa-solid fa-calendar fa-sm"></i> 2025 </a>   ·   <a href="/blog/tag/iq-tree2"> <i class="fa-solid fa-hashtag fa-sm"></i> IQ-TREE2</a>   <a href="/blog/tag/maximumlikelihood"> <i class="fa-solid fa-hashtag fa-sm"></i> MaximumLikelihood</a>   <a href="/blog/tag/substitutionmodel"> <i class="fa-solid fa-hashtag fa-sm"></i> SubstitutionModel</a>   ·   <a href="/blog/category/phylogenetics"> <i class="fa-solid fa-tag fa-sm"></i> Phylogenetics</a> </p> </header> <article class="post-content"> <div id="markdown-content"> <p>Aproveitando a aula da disciplina de <strong>Biogeografia</strong>, hoje faremos um tutorial para construção de uma árvore de máxima verossimilhança e, para isso, vamos utilizar o programa <a href="https://https://iqtree.github.io/" rel="external nofollow noopener" target="_blank">IQ-TREE3</a>. Apesar de haver alguns software que são normalmente utilizados para inferência baseada em máxima verossimilhança, este é um dos que eu mais gosto pois é um programa rápido e bastante completo. Antes de começar, faça o download do programa no <a href="https://https://iqtree.github.io/" rel="external nofollow noopener" target="_blank">website</a>, de acordo com seu sistema operacional. Neste tutorial, usaremos Windows 11 e, se você usa o mesmo sistema e quer ir direto ao ponto, pode baixá-lo <a href="https://github.com/iqtree/iqtree3/releases/download/v3.0.1/iqtree-3.0.1-Windows.zip" rel="external nofollow noopener" target="_blank">aqui</a>.</p> <p>Primeiramente, precisamos de um alinhamento. Vamos utilizar a matriz de um <a href="/assets/files/p_rupicola.fasta">fragmento de mtDNA</a> (ou DNA mitocondrial) do gene 16S, do trabalho em que <a href="https://bioone.org/journals/journal-of-herpetology/volume-54/issue-2/19-114/A-New-Rupicolous-Species-of-the-Pristimantis-conspicillatus-Group-Anura/10.1670/19-114.short" rel="external nofollow noopener" target="_blank">Taucce et al. (2020)</a> descrevem o anuro <em>Pristimantis rupicola</em>. A matriz já está alinhada e pronta para ser utilizada. Agora, descompacte o arquivo que você baixou do site do <strong>IQ-TREE2</strong> e coloque a matriz <em>p_rupícola.fasta</em> dentro da pasta com os arquivos descompactados, em uma pasta com o programa propriamente dito, chamada <em>bin</em>.</p> <p>Depois de tudo pronto, a primeira coisa que vamos fazer é clicar no programa, que está na pasta <em>bin</em>. Clique na versão do programa chamada <strong>iqtree3-click.exe</strong>. Você verá uma janela assim:</p> <div class="image-container mt-3"> <figure id="fig1"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/posts/2025-05-28/fig1.png" sizes="95vw"></source> <img src="/assets/img/posts/2025-05-28/fig1.png" class="img-fluid rounded z-depth-1" width="100%" height="auto" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> <figcaption style="font-size: 0.9em; text-align: center;"><strong>Figura 1:</strong> Primeira janela do programa IQ-TREE3 na versão de clicar</figcaption> </figure> </div> <p>Digite <em>y</em> e aperte enter. Você verá uma mensagem pedindo para que possamos inserir os parâmetros. Para realizar uma análise de máxima verossimilhança, primeiramente precisaremos inferir o melhor modelo de substituição nucleotídica. No entanto, o IQ-TREE pode fazer isso de forma automática, logo antes da busca pela árvore mais verossímil. Isso salva um monte de tempo, então vamos usar! Faremos então a busca pelo modelo de substituição nucleotídica que mais se encaixa nos dados, depois a busca pela árvore mais verossímil seguida de 1000 réplicas de bootstrap, para que possamos saber o nível de suporte dos nossos clados. Para isso, vamos digitar um comando de cada vez. Primeiro, <em>-s</em>, e então, enter. Esse primeiro comando significa que vamos fazer a busca pelo melhor modelo e então pela melhor árvore. Aperte <em>e</em> de extend, enter e então escreva o nome do arquivo, que está na pasta, no nosso caso <em>p_rupicola.fasta</em>, e então enter de novo. Depois digite <em>e</em>, aperte enter e escreva <em>-B</em>, comando para as réplicas de bootstrap ultrarápido, implementado no IQ-TREE, e aperte enter. Depois de apertar <em>e</em> novamente, digite o número de réplicas de bootstrap: 1000 está de bom tamanho. Aperte enter e então sua tela deve estar assim:</p> <div class="image-container mt-3"> <figure id="fig2"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/posts/2025-05-28/fig2.png" sizes="95vw"></source> <img src="/assets/img/posts/2025-05-28/fig2.png" class="img-fluid rounded z-depth-1" width="100%" height="auto" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> <figcaption style="font-size: 0.9em; text-align: center;"><strong>Figura 2:</strong> Janela do IQ-TREE3 depois de todos os comandos digitados</figcaption> </figure> </div> <p>Se tudo deu certo, confirme apertando <em>y</em> e então enter. Sua análise vai começar a rodar! Depois de tudo acabado, é só fechar a janela e dar uma olhada na pasta. Dentre os vários arquivos, está a nossa árvore: <em>p_rupicola.fasta.treefile</em>. Pronto! Agora é só usar um visualizador de árvore como por exemplo o <a href="https://github.com/rambaut/figtree/releases" rel="external nofollow noopener" target="_blank">FigTree</a> e comparar os resultados com a minha árvore!</p> <div class="image-container mt-3"> <figure id="fig2"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/posts/2025-05-28/fig3.png" sizes="95vw"></source> <img src="/assets/img/posts/2025-05-28/fig3.png" class="img-fluid rounded z-depth-1" width="100%" height="auto" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> <figcaption style="font-size: 0.9em; text-align: center;"><strong>Figura 3:</strong> Árvore de máxima verossimilhança seguida de 1000 réplicas de bootstrap inferida no programa IQ-TREE3</figcaption> </figure> </div> </div> </article> <br> <hr> <br> <ul class="list-disc pl-8"></ul> <h2 class="text-3xl font-semibold mb-4 mt-12">Enjoy Reading This Article?</h2> <p class="mb-2">Here are some more articles you might like to read next:</p> <li class="my-2"> <a class="text-pink-700 underline font-semibold hover:text-pink-800" href="/blog/2024/calibrated_gene_tree/">Building a calibrated gene tree with BEAST 2.7</a> </li> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2025 Pedro P. G. Taucci. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. Last updated: May 28, 2025. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?e0514a05c5c95ac1a93a8dfd5249b92e"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6" crossorigin="anonymous"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> <script src="/assets/js/vanilla-back-to-top.min.js?f40d453793ff4f64e238e420181a1d17"></script> <script>addBackToTop();</script> <script type="module" src="/assets/js/search/ninja-keys.min.js?a3446f084dcaecc5f75aa1757d087dcf"></script> <ninja-keys hidebreadcrumbs noautoloadmdicons placeholder="Type to start searching"></ninja-keys> <script>let searchTheme=determineComputedTheme();const ninjaKeys=document.querySelector("ninja-keys");"dark"===searchTheme?ninjaKeys.classList.add("dark"):ninjaKeys.classList.remove("dark");const openSearchModal=()=>{const e=$("#navbarNav");e.hasClass("show")&&e.collapse("hide"),ninjaKeys.open()};</script> <script>const ninja=document.querySelector("ninja-keys");ninja.data=[{id:"nav-home",title:"Home",section:"Navigation",handler:()=>{window.location.href="/"}},{id:"nav-blog",title:"Blog",description:"",section:"Navigation",handler:()=>{window.location.href="/blog/"}},{id:"nav-publications",title:"Publications",description:"Publications by categories in reversed chronological order. Generated by jekyll-scholar.",section:"Navigation",handler:()=>{window.location.href="/publications/"}},{id:"nav-projects",title:"Projects",description:"Explore our collection of research initiatives and science communication projects.",section:"Navigation",handler:()=>{window.location.href="/projects/"}},{id:"nav-repositories",title:"Repositories",description:"GitHub repositories of LaBESTA and its members.",section:"Navigation",handler:()=>{window.location.href="/repositories/"}},{id:"nav-cv",title:"CV",description:"This is a summarized version of my CV. For publications, please go to the Publications tab, and if you&#39;d like a more complete version, click the button above.",section:"Navigation",handler:()=>{window.location.href="/cv/"}},{id:"nav-teaching",title:"Teaching",description:"Materials for undergraduate and graduate courses taught.",section:"Navigation",handler:()=>{window.location.href="/teaching/"}},{id:"nav-people",title:"People",description:"Members of LaBESTA",section:"Navigation",handler:()=>{window.location.href="/people/"}},{id:"post-construindo-uma-\xe1rvore-de-m\xe1xima-verossimilhan\xe7a-com-o-software-iq-tree3",title:"Construindo uma \xe1rvore de m\xe1xima verossimilhan\xe7a com o software IQ-TREE3",description:"A step-by-step beginner&#39;s guide",section:"Posts",handler:()=>{window.location.href="/blog/2025/maximum_likelihood_tree/"}},{id:"post-building-a-calibrated-gene-tree-with-beast-2-7",title:"Building a calibrated gene tree with BEAST 2.7",description:"A step-by-step beginner&#39;s guide",section:"Posts",handler:()=>{window.location.href="/blog/2024/calibrated_gene_tree/"}},{id:"news-we-just-arrived-to-s\xe3o-paulo-state-university-unesp-at-bauru-s\xe3o-paulo",title:"We just arrived to S\xe3o Paulo State University (Unesp) at Bauru, S\xe3o Paulo!...",description:"",section:"News"},{id:"projects-drag\xf5es-de-garagem",title:"Drag\xf5es de Garagem",description:"Bringing science to the public through engaging discussions and accessible content as part of one of Brazil\u2019s top science outreach podcasts",section:"Projects",handler:()=>{window.location.href="/projects/DdG/"}},{id:"projects-systematics-of-ischnocnema",title:"Systematics of Ischnocnema",description:"Taxonomic review and molecular phylogenetics of the Neotropical frog genus Ischnocnema (Anura: Brachycephaloidea)",section:"Projects",handler:()=>{window.location.href="/projects/systematics_of_ischnocnema/"}},{id:"teaching-taxonomia-integrativa",title:"Taxonomia Integrativa",description:"Material da disciplina Taxonomia Integrativa",section:"Teaching",handler:()=>{window.location.href="/teaching/taxonomia_integrativa/"}},{id:"teaching-vertebrados-ii",title:"Vertebrados II",description:"Material da disciplina de Vertebrados II, Unesp Bauru",section:"Teaching",handler:()=>{window.location.href="/teaching/vertebrados2/"}},{id:"socials-email",title:"Send email",section:"Socials",handler:()=>{window.open("mailto:%70%65%64%72%6F.%74%61%75%63%63%69@%75%6E%65%73%70.%62%72","_blank")}},{id:"socials-orcid",title:"ORCID",section:"Socials",handler:()=>{window.open("https://orcid.org/0000-0002-3088-4543","_blank")}},{id:"socials-google-scholar",title:"Google Scholar",section:"Socials",handler:()=>{window.open("https://scholar.google.com/citations?user=67gGXtMAAAAJ","_blank")}},{id:"socials-lattes",title:"Lattes",section:"Socials",handler:()=>{window.open("http://lattes.cnpq.br/6010051338306653","_blank")}},{id:"socials-researchgate",title:"ResearchGate",section:"Socials",handler:()=>{window.open("https://www.researchgate.net/profile/Pedro-Taucce/","_blank")}},{id:"socials-github",title:"GitHub",section:"Socials",handler:()=>{window.open("https://github.com/labesta","_blank")}},{id:"socials-x",title:"X",description:"Twitter",section:"Socials",handler:()=>{window.open("https://twitter.com/pedrotaucce","_blank")}},{id:"socials-instagram",title:"Instagram",section:"Socials",handler:()=>{window.open("https://instagram.com/labesta_unesp","_blank")}},{id:"light-theme",title:"Change theme to light",description:"Change the theme of the site to Light",section:"Theme",handler:()=>{setThemeSetting("light")}},{id:"dark-theme",title:"Change theme to dark",description:"Change the theme of the site to Dark",section:"Theme",handler:()=>{setThemeSetting("dark")}},{id:"system-theme",title:"Use system default theme",description:"Change the theme of the site to System Default",section:"Theme",handler:()=>{setThemeSetting("system")}}];</script> <script src="/assets/js/shortcut-key.js?6f508d74becd347268a7f822bca7309d"></script> </body> </html>